### PR TITLE
Refactored Laminar

### PR DESCRIPTION
**Pull request by TimonPost**
- Moved some codes to reduce the number of folders.
   - /error => /
   - /net/connection => /net
- Refactored some documentation comments to be more correct.
- Updated README.
- Removed some public types which should supose to be public.
- Removed unused Errors from `ErrorKind`

For the readme and in the code I want to link to our book, however, I can't do that yet, and can be done later if #165 is merged and live.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,10 +15,26 @@
 [s6]: https://tokei.rs/b1/github/amethyst/laminar?category=code
 [s7]: https://codecov.io/gh/amethyst/laminar/branch/master/graphs/badge.svg
 
-This library implements some TCP-like features on top of an UDP-socket. 
-It will provide a lightweight message-based interface with certain guarantees like reliability, fragmentation congestion monitoring.
+Laminar is a semi-reliable UDP-based protocol for multiplayer games. This library implements wrappers around the UDP-protocol,
```
Comment (LucioFranco) [inline README.md]: `This library implements wrappers around a UDP socket`

File: README.md [hunk 2]
```diff
@@ -35,16 +51,15 @@ Laminar was designed to be used in the [Amethyst][amethyst] game engine and is l
...
+- Connection Tracking
 - Automatic Fragmentation
-- Unreliable and Reliable packets
-- Protocol versioning
-- RTT estimation
-- Link conditioner to simulate packet loss and latency
-- Well tested by integration tests and unit tests
-- Good error handling
-- Benchmarks
+- Packets could be sent: Unreliable: Sequenced, Reliable: Unordered, Sequenced, Ordered.
```
Comment (LucioFranco) [inline README.md]: `Packet types:`

File: README.md [hunk 3]
```diff
@@ -133,15 +147,12 @@ match result {
 - [Lucio Franco](https://github.com/LucioFranco)
 - [Fletcher Haynes](https://github.com/fhaynes)
 - [Timon Post](https://github.com/TimonPost)
-- [Justin LeFebvre](https://github.com/jstnlef)
-
-We want to give credit to [gaffer on games](https://gafferongames.com/) as we have used his guide to building a game networking protocol to build this library.
```
Comment (LucioFranco) [inline README.md]: why is this removed?

File: src/config.rs [hunk 4]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
 use std::{default::Default, time::Duration};
 
 #[derive(Clone)]
-/// Struct that contains config values for various aspects of the network
+/// Type that contains the configuration options to configure laminar for special use-cases.
```
Comment (LucioFranco) [inline src/config.rs]: you can just drop type that, `Contains the configuration....`

File: src/config.rs [hunk 5]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
...
 pub struct Config {
-    /// The maximal amount of time to keep `VirtualConnection`s around before cleaning them up.
+    /// The maximal amount of time to keep connection alive before cleaning them up.
```
Comment (LucioFranco) [inline src/config.rs]: `keep the connection`

File: src/config.rs [hunk 6]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
...
     pub idle_connection_timeout: Duration,
     /// This is the maximal size a packet can get with all its fragments.
     ///
     /// Recommended value: 16384
     pub max_packet_size: usize,
-    /// These are the maximal fragments a packet could be divided into.
+    /// This is the value which specifies the maximal allowed fragments.
```
Comment (LucioFranco) [inline src/config.rs]: I find it better here again to just say `Value which can specify...`, since we know what this is by looking down :)

Not sure if this is the right pattern but imo I think shorter top descriptions is better.

File: src/error.rs [hunk 7]
```diff
@@ -1,11 +1,144 @@
-///! Module contains error handling and error types for laminar
-mod error_kinds;
-mod network_error;
-
-pub use self::error_kinds::{FragmentErrorKind, PacketErrorKind};
-pub use self::network_error::ErrorKind;
-
-use std::result;
+///! This module contains the laminar error handling logic.
```
Comment (LucioFranco) [inline src/error.rs]: two nits: this should be `//!`

second, we should have a space between the module doc and the first line of code in it. Right now it makes it look like you are adding a doc comment for `use crate...`

File: src/error.rs [hunk 8]
```diff
@@ -1,11 +1,144 @@
...
+            FragmentErrorKind::AlreadyProcessedFragment => {
+                write!(fmt, "The fragment received was already processed.")
+            }
+            FragmentErrorKind::FragmentWithUnevenNumberOfFragemts => write!(
+                fmt,
+                "The fragment header does not contain the right fragment count."
+            ),
+            FragmentErrorKind::CouldNotFindFragmentById => write!(
+                fmt,
+                "The fragment supposed to be in a cache but it was not found."
```
Comment (LucioFranco) [inline src/error.rs]: `The fragment supposed to be in a the cache but it was not found.`

File: src/lib.rs [hunk 9]
```diff
@@ -1,49 +1,41 @@
-//! Laminar semi-reliable UDP protocol for multiplayer games. This library implements wraps around a UDP
-//! and provides light weight stream based interface that provides certain guarantees like reliability.
+//! Laminar is a semi-reliable UDP-based protocol for multiplayer games. This library implements wrappers around the UDP-protocol,
```
Comment (LucioFranco) [inline src/lib.rs]: we should align this with the readme changes, which it seems you have but just leaving this to remind you to make one last check here.

File: src/lib.rs [hunk 10]
```diff
@@ -1,49 +1,41 @@
...
 //! [readme]: https://github.com/amethyst/laminar/blob/master/README.md
-//!
-//! # Example
-//!
+//! [book]: https://github.com/amethyst/laminar/tree/master/docs/md_book
+//! [docs]: https://docs.rs/laminar/
+//! [examples]: https://github.com/amethyst/laminar/tree/master/examples
 
 #![warn(missing_docs)]
 #![allow(clippy::trivially_copy_pass_by_ref)]
```
Comment (LucioFranco) [inline src/lib.rs]:
```suggestion
#![allow(clippy::trivially_copy_pass_by_ref)]

```

File: src/net/events.rs [hunk 11]
```diff
@@ -1,16 +1,14 @@
 use crate::packet::Packet;
 use std::net::SocketAddr;
 
-/// Events which will be pushed through the event_receiver returned by `Socket::bind`.
+/// Events that could occur in laminar, those will be pushed through the `event_receiver` returned by `Socket::bind`.
```
Comment (LucioFranco) [inline src/net/events.rs]:
```suggestion
/// Events that can occur in `laminar` and that will be pushed through the `event_receiver` returned by `Socket::bind`.
```

File: src/net/events.rs [hunk 12]
```diff
@@ -1,16 +1,14 @@
...
 #[derive(Debug, PartialEq)]
 pub enum SocketEvent {
-    /// A packet has been received from a client.
+    /// A packet was received from a client.
     Packet(Packet),
-    /// A new client connects. Clients are uniquely identified by the ip:port combination at this layer.
+    /// A new client connected.
```
Comment (LucioFranco) [inline src/net/events.rs]:
```suggestion
    /// A new client connected.
    
```

File: src/packet.rs [hunk 13]
```diff
@@ -1,4 +1,4 @@
-/// Contains code dealing with Packet Headers
+///! This module provides all the logic around the packet, reading, parsing, constructing headers etc.
```
Comment (LucioFranco) [inline src/packet.rs]:
```suggestion
//! This module provides all the logic around the packet, reading, parsing, constructing headers etc.

```

File: README.md (hunk 3 above)
Comment (TimonPost) [reply README.md]: Because we already said in the header of the read me that we had some ideas from him. However, there are millions of other resources we could be referencing here.

Comment (LucioFranco) [reply README.md]: Ah okay, yeah as long as we give credit to atleast what inspired us id like that.

File: src/error.rs (hunk 7 above)
Comment (TimonPost) [reply src/error.rs]: ow, must have slipped through.

File: README.md (hunk 1 above)
Comment (TimonPost) [reply README.md]: both are correct, but we are not wrapping around a socket perse but more around the protocol itself.

Comment (LucioFranco) [reply README.md]: well technically we are implementing a udp protocol on top of a udp socket protocol is already in udp via the p.

Comment (TimonPost) [reply README.md]: Technically speaking David P. Reed implemented the UDP-protocol. This protocol was defined way back in 1980.

File: README.md [hunk 14]
```diff
@@ -35,16 +51,15 @@ Laminar was designed to be used in the [Amethyst][amethyst] game engine and is l
...
+- Connection Tracking
 - Automatic Fragmentation
-- Unreliable and Reliable packets
-- Protocol versioning
-- RTT estimation
-- Link conditioner to simulate packet loss and latency
-- Well tested by integration tests and unit tests
-- Good error handling
-- Benchmarks
+- Packets Types: Unreliable: Sequenced, Reliable: Unordered, Sequenced, Ordered.
```
Comment (fhaynes) [inline README.md]:
```suggestion
- Packets Types: Unreliable and Sequenced, Reliable and Unordered, Sequenced and Ordered.
```

File: README.md [hunk 15]
```diff
@@ -35,16 +51,15 @@ Laminar was designed to be used in the [Amethyst][amethyst] game engine and is l
...
+- Arranging Streams
+- Protocol Visioning
```
Comment (fhaynes) [inline README.md]: What's `Visioning`?

File: README.md [hunk 16]
```diff
@@ -35,16 +51,15 @@ Laminar was designed to be used in the [Amethyst][amethyst] game engine and is l
...
+- RTT-Estimation
```
Comment (fhaynes) [inline README.md]:
```suggestion
- RTT Estimation
```

File: README.md [hunk 17]
```diff
@@ -35,16 +51,15 @@ Laminar was designed to be used in the [Amethyst][amethyst] game engine and is l
...
+- Link conditioner to simulate packet loss and latency.
+- Well tested by integration and unit tests.
```
Comment (fhaynes) [inline README.md]:
```suggestion
- Well-tested by integration and unit tests
```

File: README.md [hunk 18]
```diff
@@ -35,16 +51,15 @@ Laminar was designed to be used in the [Amethyst][amethyst] game engine and is l
...
-- RTT estimation
-- Link conditioner to simulate packet loss and latency
-- Well tested by integration tests and unit tests
-- Good error handling
-- Benchmarks
+- Packets Types: Unreliable: Sequenced, Reliable: Unordered, Sequenced, Ordered.
+- Arranging Streams
+- Protocol Visioning
+- RTT-Estimation
+- Link conditioner to simulate packet loss and latency.
```
Comment (fhaynes) [inline README.md]:
```suggestion
- Link conditioner to simulate packet loss and latency
```

File: README.md [hunk 19]
```diff
@@ -133,15 +147,12 @@ match result {
 - [Lucio Franco](https://github.com/LucioFranco)
 - [Fletcher Haynes](https://github.com/fhaynes)
 - [Timon Post](https://github.com/TimonPost)
-- [Justin LeFebvre](https://github.com/jstnlef)
```
Comment (fhaynes) [inline README.md]: We should leave Justin in the credits I think?

File: README.md [hunk 20]
```diff
@@ -133,15 +147,12 @@ match result {
...
-
-We want to give credit to [gaffer on games](https://gafferongames.com/) as we have used his guide to building a game networking protocol to build this library. 
+- [Justin LeFebvre](https://github.com/jstnlef) 
 
 ## Note
 
-This library is not fully stable yet. 
-Although version 0.1.0 is released we might have to change some of the existing API.  
-Laminar is used in [Amethyst-Network](https://github.com/amethyst/amethyst/tree/master/amethyst_network), you could give that a look if you want to see some more advanced use-cases.
+This library is not fully stable yet, and changes to the API could be expected.
```
Comment (fhaynes) [inline README.md]:
```suggestion
This library is not fully stable yet, and there may be breaking changes to the API.
```

File: README.md [hunk 21]
```diff
@@ -133,15 +147,12 @@ match result {
...
-We want to give credit to [gaffer on games](https://gafferongames.com/) as we have used his guide to building a game networking protocol to build this library. 
+- [Justin LeFebvre](https://github.com/jstnlef) 
 
 ## Note
 
-This library is not fully stable yet. 
-Although version 0.1.0 is released we might have to change some of the existing API.  
-Laminar is used in [Amethyst-Network](https://github.com/amethyst/amethyst/tree/master/amethyst_network), you could give that a look if you want to see some more advanced use-cases.
+This library is not fully stable yet, and changes to the API could be expected.  
+For more advanced use-cases of laminar, you could check out the [Amethyst-Network](https://github.com/amethyst/amethyst/tree/master/amethyst_network) crate.
```
Comment (fhaynes) [inline README.md]:
```suggestion
For more advanced examples of using laminar, you can check out the [Amethyst-Network](https://github.com/amethyst/amethyst/tree/master/amethyst_network) crate.
```

File: src/config.rs [hunk 22]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
 use std::{default::Default, time::Duration};
 
 #[derive(Clone)]
-/// Struct that contains config values for various aspects of the network
+/// Contains the configuration options to configure laminar for special use-cases.
 pub struct Config {
-    /// The maximal amount of time to keep `VirtualConnection`s around before cleaning them up.
+    /// Value which can specify the amount of time to keep the connection alive before cleaning them up.
```
Comment (fhaynes) [inline src/config.rs]:
```suggestion
    /// Value which can specify the amount of time that can pass without hearing from a client before considering them disconnected
```

File: src/config.rs [hunk 23]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
...
     pub idle_connection_timeout: Duration,
-    /// This is the maximal size a packet can get with all its fragments.
+    /// Value which can specify the maximal size a packet can get with all its fragments.
```
Comment (fhaynes) [inline src/config.rs]:
```suggestion
    /// Value which can specify the maximum size a packet can be in bytes. This value is inclusive of fragmenting; if a packet is fragmented, the total size of the fragments cannot exceed this value.
```

File: src/config.rs [hunk 24]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
...
     pub max_packet_size: usize,
-    /// These are the maximal fragments a packet could be divided into.
+    /// Value which can specify the maximal allowed fragments.
     ///
     /// Why can't I have more than 255 (u8)?
     /// This is because you don't want to send more then 256 fragments over UDP, with high amounts of fragments the chance for an invalid packet is very high.
     /// Use TCP instead (later we will probably support larger ranges but every fragment packet then needs to be resent if it doesn't get an acknowledgement).
     ///
-    /// Recommended value: 16 but keep in mind that lower is better.
+    /// default: 16 but keep in mind that lower is better.
```
Comment (fhaynes) [inline src/config.rs]: The maximum number of fragments into which a packet can be split. Defaults to 16. Keep in mind lower is better.

File: src/config.rs [hunk 25]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
...
     pub max_fragments: u8,
-    /// This is the size of a fragment.
-    /// If a packet is too large it needs to be split in fragments.
+    /// Value which can specify the size of a fragment.
     ///
-    /// Recommended value: +- 1450 (1500 is the default MTU)
+    /// use-case: If a packet is too large it needs to be split into fragments.
```
Comment (fhaynes) [inline src/config.rs]:
```suggestion
    /// This is the maximum size of each fragment. It defaults to `1450` bytes, due to the default MTU on most network devices being `1500`. 
```

File: src/config.rs [hunk 26]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
...
+    /// default: +- 1450 (1500 is the default MTU)
```
Comment (fhaynes) [inline src/config.rs]:
```suggestion
```

File: src/config.rs [hunk 27]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
...
     pub fragment_size: u16,
-    /// This is the size of the buffer that queues up fragments ready to be reassembled once the whole packet arrives.
+    /// Value which can specify size of the buffer that queues up fragments ready to be reassembled once the whole packet arrives.
```
Comment (fhaynes) [inline src/config.rs]:
```suggestion
    /// Value which can specify the size of the buffer that queues up fragments ready to be reassembled once all fragments have arrived.```

File: src/config.rs [hunk 28]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
...
     pub fragment_reassembly_buffer_size: usize,
-    /// This is the size of the buffer the UDP socket reads it data into.
+    /// Value which can specify size of the buffer the UDP-bytes will be read into.
+    ///
+    /// default: +- 1450 (1500 is the default MTU)
```
Comment (fhaynes) [inline src/config.rs]:
```suggestion
    /// Value that specifies the size of the buffer the UDP data will be read into. Defaults to `1450` bytes.
```

Should probably clarify if this is or is not setting things at the OS level

File: src/config.rs [hunk 29]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
...
     ///
-    /// Recommended value: +- 1450 (1500 is the default MTU)
+    /// use-case: If a packet is too large it needs to be split into fragments.
+    /// default: +- 1450 (1500 is the default MTU)
     pub fragment_size: u16,
-    /// This is the size of the buffer that queues up fragments ready to be reassembled once the whole packet arrives.
+    /// Value which can specify size of the buffer that queues up fragments ready to be reassembled once the whole packet arrives.
     pub fragment_reassembly_buffer_size: usize,
-    /// This is the size of the buffer the UDP socket reads it data into.
+    /// Value which can specify size of the buffer the UDP-bytes will be read into.
```
Comment (fhaynes) [inline src/config.rs]:
```suggestion
```

File: src/config.rs [hunk 30]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
...
+    ///
```
Comment (fhaynes) [inline src/config.rs]:
```suggestion
```

File: src/config.rs [hunk 31]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
...
+    /// default: +- 1450 (1500 is the default MTU)
     pub receive_buffer_max_size: usize,
-    /// This is the factor which will smooth out network jitter. So that if one packet is not arrived fast we don't wan't to directly transform to an bad network.
+    /// Value which can specify the factor which will smooth out network jitter.
     ///
-    /// Recommended value: 10% of the rtt time.
-    /// Value is a ratio (0 = 0% and 1 = 100%)
+    /// use-case: If one packet hast not arrived we don't directly want to transform to a bad network state.
+    /// default: 10% of the rtt time (ratio (0 = 0% and 1 = 100%)
```
Comment (fhaynes) [inline src/config.rs]:
```suggestion
/// Value that specifies the factor used to smooth out network jitter. It defaults to 10% of the round-trip time. It is expressed as a ratio, with 0 equal to 0% and 1 equal to 100%. This helps prevent flapping of `VirtualConnections`
```

File: src/config.rs [hunk 32]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
...
     pub rtt_smoothing_factor: f32,
-    /// This is the maximal round trip time (rtt) for packet.
+    /// Value which can specify the maximal round trip time (rtt) for packet.
     ///
-    /// Recommend value: 250 ms
-    /// Value is represented in milliseconds.
+    /// default: 250 ms
```
Comment (fhaynes) [inline src/config.rs]:
```suggestion
    /// Value which specifies the maximum round trip time before we consider it a problem. This is expressed in milliseconds.
```

File: src/config.rs [hunk 33]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
...
     pub rtt_max_value: u16,
-    /// This is the size of the event buffer we read socket events from `mio::Poll` into.
+    /// Value which can specify the event buffer we read socket events into.
+    ///
+    /// default: 1024
```
Comment (fhaynes) [inline src/config.rs]:
```suggestion
    /// Value that specifies the size of the event buffer into which we receive socket events, in bytes. Defaults to 1024.
```

File: src/config.rs [hunk 34]
```diff
@@ -2,44 +2,49 @@ use crate::net::constants::{DEFAULT_MTU, FRAGMENT_SIZE_DEFAULT, MAX_FRAGMENTS_DE
...
     pub socket_event_buffer_size: usize,
-    /// Optional duration specifying how long we should block polling for socket events.
+    /// Value which can specify how long we should block polling for socket events.
+    ///
+    /// default: 1 ms
```
Comment (fhaynes) [inline src/config.rs]:
```suggestion
    /// Value that specifies how long we should block polling for socket events, in milliseconds. Defaults to `1ms`.
```
Should probably add in what is being polled.

File: src/error.rs [hunk 35]
```diff
@@ -1,11 +1,145 @@
...
+            ErrorKind::CouldNotReadHeader(header) => write!(
+                fmt,
+                "Expected {} header but could not be read from buffer.",
+                header
+            ),
+        }
+    }
+}
+
+/// Errors that could occur with constructing parsing packet contents
```
Comment (fhaynes) [inline src/error.rs]:
```suggestion
/// Errors that could occur while parsing packet contents
```

File: src/lib.rs [hunk 36]
```diff
@@ -1,49 +1,42 @@
...
 //! # Concepts
 //!
-//! This library is mostly based off of [Gaffer on Games][gog] and shares features with RakNet. The idea is to provide a low level
-//! UDP protocol that supports the use cases of video games that require multilayer features. The library
-//! itself provides a few low level types of packets that provides different types of guarantees. The most
-//! basic are unreliable and reliable packets. This generally correlates to state update packets that do not
-//! require to be synced, meaning the packet can get dropped without harm to the game. The other is used for
-//! example score updates, where you want to make sure that the data is received on the other end even in case
-//! of a packet drop. For more information, read the projects [README.md][readme]
+//! This library is loosely based off of [Gaffer on Games][gog] and shares features similar as RakNet, Steam Socket, netcode.io.
```
Comment (fhaynes) [inline src/lib.rs]:
```suggestion
//! This library is loosely based off of [Gaffer on Games][gog] and has features similar to RakNet, Steam Socket, and netcode.io.
```

File: src/lib.rs [hunk 37]
```diff
@@ -1,49 +1,42 @@
...
+//! The idea is to provide an in rust written, low-level UDP-protocol which supports the use of cases of video games that require multilayer features.
```
Comment (fhaynes) [inline src/lib.rs]:
```suggestion
//! The idea is to provide a native Rust low-level UDP-protocol which supports the use of cases of video games that require multiplayer features.
```

File: src/lib.rs [hunk 38]
```diff
@@ -1,49 +1,42 @@
...
+//! The library itself provides a few low-level types of packets that provide different types of guarantees. The most
+//! basic are unreliable and reliable packets. Also ordering, sequencing can be done on multiple streams.
```
Comment (fhaynes) [inline src/lib.rs]:
```suggestion
//! basic are unreliable and reliable packets. Ordering, sequencing can be done on multiple streams.
```

File: src/net/events.rs [hunk 39]
```diff
@@ -1,16 +1,15 @@
...
+
+    /// Clients are uniquely identified by the ip:port combination at this layer.
     Connect(SocketAddr),
-    /// A client disconnects. This is generated from the server-side intentionally disconnecting a client,
-    /// or it could be from the client disconnecting.
-    Disconnect(SocketAddr),
-    /// This is generated if the server has not seen traffic from a client after a configurable amount of time.
+    /// The client has been idling for a configurable amount of time.
```
Comment (fhaynes) [inline src/net/events.rs]: Should specify which config variable affects this, I think

File: src/packet.rs [hunk 40]
```diff
@@ -1,4 +1,5 @@
-/// Contains code dealing with Packet Headers
+//! This module provides all the logic around the packet, reading, parsing, constructing headers etc.
```
Comment (fhaynes) [inline src/packet.rs]:
```suggestion
//! This module provides all the logic around the packet, such as reading, parsing, and constructing headers.
```

File: src/packet/header.rs [hunk 41]
```diff
@@ -1,3 +1,6 @@
+//! This module provides parses and readers for the headers that could be appended to any packet.
+//! We use headers to control reliability, fragmentation, ordering.
```
Comment (fhaynes) [inline src/packet/header.rs]:
```suggestion
//! We use headers to control reliability, fragmentation, and ordering.
```

File: README.md (hunk 15 above)
Comment (TimonPost) [reply README.md]: * Versioning

File: README.md (hunk 14 above)
Comment (TimonPost) [reply README.md]: Now we cant have: unreliable only, or unreliable sequenced. This is not correct we have the following types:

Reliable + Ordered
Reliable + Unordered
Reliable + Sequenced
Unreliable
Unreliable + Sequenced.

File: README.md (hunk 19 above)
Comment (TimonPost) [reply README.md]: He still is, check out the lines above that.

Comment (jstnlef) [reply README.md]: Weird diff 👍. All good though.

**Comment by TimonPost** [comment]
bors r=LucioFranco,jstnlef,fhaynes
